File: src/main/java/com/polykhel/billnext/repository/WalletRepository.java
package com.polykhel.billnext.repository;

import com.polykhel.billnext.domain.Wallet;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

/**
 * Spring Data SQL repository for the Wallet entity.
 */
@Repository
public interface WalletRepository extends JpaRepository<Wallet, Long>, JpaSpecificationExecutor<Wallet> {}

File: src/main/kotlin/com/polykhel/billnext/repository/WalletRepository.kt
package com.polykhel.billnext.repository

import com.polykhel.billnext.domain.Wallet
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

/**
 * Spring Data SQL repository for the [Wallet] entity.
 */
@Suppress("unused")
@Repository
interface WalletRepository : JpaRepository<Wallet, Long>, JpaSpecificationExecutor<Wallet> {

    @Query("select wallet from Wallet wallet where wallet.walletGroup.user.login = ?#{principal.username}")
    fun findUserIsCurrentUser(pageable: Pageable): Page<Wallet>
}
